### PR TITLE
Issues 2036

### LINKED ISSUE #2036: Add setting to select default state
## Issue Overview

Right now we can set the default country. But we should also allow admins to select the default _state_ if applicable.

For instance, if I run a nonprofit that's specific to California I would want the state to always be CA by default. This would allow easier donations for state residents because they wouldn't have to select their state. 

## Expected Behavior
Admins can select the default state in general settings.

## Current Behavior
There is no option to select the default state.

## Possible Solution
Create a new field below "Base country" that is "Default state/province" and have this field conditional based on the base country. If you change from the US for instance to another country with states/provides/whatever then it will update accordingly.

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
Admin can select default state/province (if any) while selecting/changing the country.
This PR will resolve #2036

## How Has This Been Tested?
Manually tested
Please view the video for the step 
http://screencast-o-matic.com/watch/cbQh6nI1m6 ( updated on 09.05.2017 )

## Types of changes
Bug fix (non-breaking change which fixes an issue)

Video Link
http://screencast-o-matic.com/watch/cbQh6nI1m6 ( updated on 09.05.2017 )

## Checklist:
- [x] PHPCS error.
- [x] PPH Compatibility.
- [ ] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by ravinderk** [comment]
@raftaar1191 It is not working when you reset country.

okokkp4l3a

**Comment by raftaar1191** [comment]
@ravinderk @DevinWalker 

Fixed all the issues and also add the video that covers all the test case where the default states are being used

**Comment by DevinWalker** [comment]
Thanks @raftaar1191 the video was good. 

I have a few questions:

1. One thing I didn't see though was if you tested the `[give_profile_editor]` shortcode on the frontend. Actually, when I tested with it I noticed a bug with the state field not being updated entirely. Can you fix that too while also ensuring it works accordingly? 



2. What will happen when users update with the new state field? For backwards compatibility it should just be blank. I don't think we want to presume to set a state for them.

**Comment by raftaar1191** [comment]
@DevinWalker 

Fixed all the issues

File: assets/js/admin/admin-scripts.js [hunk 1]
```diff
@@ -467,11 +467,43 @@ jQuery.noConflict();
 	var Give_Settings = {
 
 		init: function () {
+			this.setting_change_Country();
```
Comment (DevinWalker) [inline assets/js/admin/admin-scripts.js]: How come the "C" is capped here? Can you just keep them all lowercase?

Comment (raftaar1191) [reply assets/js/admin/admin-scripts.js]: Fixed

**Comment by DevinWalker** [comment]
@raftaar1191 please take care of what we reviewed with the ability to save an "empty" state field like so:

**Pull request merged**